lib/active_model_serializers/lookup_chain.rb

```diff
@@ -0,0 +1,43 @@
+module ActiveModelSerializers
+  module LookupChain
+    BY_RESOURCE = lambda do |_resource_class, resource_class_name, _serializer_class|
+      "#{resource_class_name}Serializer"
+    end
+
+    BY_NAMESPACE = lambda  do |resource_class, resource_class_name, _serializer_class|
+      resource_namespace = resource_class.name.deconstantize
+      serializer_name = "#{resource_class_name}Serializer"
```

Comment:
To remove the duplication you could do something like:

```ruby
class SerializerLookup
  class << self
    def by_namespace(lookup_namespace, klass)
      "#{resource_namespace(klass)}::#{serializer_name(klass)}"
    end

    def by_parent_serializer(lookup_namespace, klass)
      # re-use same logic
    end

    private

    def resource_namespace(klass)
      klass.name.deconstantize
    end

    def serializer_name(klass)
      "#{resource_class_name}Serializer"
    end
end

procs << SerializerLookup.method(:by_namespace)
procs << SerializerLookup.method(:by_parent_serializer)
```

That said - I think an array of procs is a good idea, but I'm not chained to it.